sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -442,17 +442,35 @@ class Analyzer(
         child: LogicalPlan): LogicalPlan = {
       val gid = AttributeReference(VirtualColumn.groupingIdName, IntegerType, false)()
 
+      // In case of ANSI-SQL compliant syntax for GROUPING SETS, groupByExprs is optional and
+      // can be null. In such case, we derive the groupByExprs from the user supplied values for
+      // grouping sets.
+      val finalGroupByExpressions = if (groupByExprs == Nil) {
+        selectedGroupByExprs.flatten.foldLeft(Seq.empty[Expression]) { (result, currentExpr) =>
```

Comment:
What if `GROUP BY GROUPING SETS (())`?

Reply:
@viirya No. We should be getting an error as we don't have a group by specification.

Reply:
Can we have a test case for it too?

Reply:
@viirya Yeah.. was already adding it .. knew u would ask :-)

Reply:
@dilipbiswal Thanks! :-)